ReactWindows/ReactNative/Views/View/ReactViewManager.cs

```diff
@@ -0,0 +1,270 @@
+
+                            items.Add(new JObject
+                            {
+                                { "kind", "file" },
+                                { "type", type }
+                            });
+
+                            types.Add(type);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        System.Diagnostics.Debug.WriteLine("GetDataTransferInfo: " + ex);
+                    }
+                }
+            }
+
+            return new JObject
+            {
+                { "files", files },
+                { "items", items },
+                { "types", types }
+            };
+        }
+
+        private static string GetAccessToken(StorageFile file)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
+            var capacity = futureAccessList.MaximumItemsAllowed;
+            var now = DateTime.Now;
+
+            // Use more aggresive timeout when number of items exceed 100 or half of the capacity
+            // Note: 100 is an arbitary limit in case that OS reduces the capacity (it is 1000 in RS1). 
+            var expiry = existingEntries.Count >= capacity / 2 || existingEntries.Count >= 100 ?
+                now.AddDays(-1) :
+                now.AddDays(-7);
+
+            foreach (var entry in existingEntries)
+            {
+                if (DateTime.TryParse(entry.Metadata, out DateTime time) && time < expiry)
+                {
+                    futureAccessList.Remove(entry.Token);
+                }
+            }
+
+            return futureAccessList.Add(file, now.ToString(System.Globalization.CultureInfo.InvariantCulture));
+        }
+
+        private static async void OnDragEnter(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+            var data = await GetDataTransferInfo(args.DataView);
+
+            view.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(new DragDropEvent(view.GetTag(), "topDragEnter", data));
+        }
+
+        private static async void OnDragOver(object sender, DragEventArgs args)
+        {
+            var view = sender as BorderedCanvas;
+
+            // In web when JS gets a "drag over" event, it modifies the event
+            // object to tell if the DOM element supports dropping items:
+            //
+            //      e.dataTransfer.effectAllowed = 'copy';
+            //      e.dataTransfer.dropEffect = 'copy';
+            //
+            // However in RN this approach doesn't work, so we use a simpler
+            // solution: JS sets allowDrop=true which implies that the RN element
+            // always allows dropping items.
+            args.AcceptedOperation = DataPackageOperation.Copy;
+
```

Comment:
[https://docs.microsoft.com/en-us/windows/uwp/app-to-app/share-data](https://docs.microsoft.com/en-us/windows/uwp/app-to-app/share-data)

Reply:
The share contract? How does it help here?